pkg/tasks/tasks.go

```diff
@@ -116,20 +116,36 @@ func WithFullInstall(t Tasks) Tasks {
 			{Fn: saveKubeconfig, ErrMsg: "failed to save kubeconfig to the local machine"},
 			{Fn: restartKubeAPIServer, ErrMsg: "failed to restart unhealthy kube-apiserver"},
 		}...).
-		append(kubernetesResources()...).
+		append(WithResources(nil)...).
 		append(
 			Task{Fn: createMachineDeployments, ErrMsg: "failed to create worker machines"},
 		)
 }
 
-func WithRefreshResources(t Tasks) Tasks {
+func WithResources(t Tasks) Tasks {
 	return t.append(
 		Tasks{
+			{
+				Fn:         renewCerts,
+				ErrMsg:     "failed to renew certificates",
+				Desciption: "renew all certificates",
+				Predicate: func(s *state.State) bool {
+					return s.LiveCluster.CertsToExpireInLessThen90Days()
+				},
```

Comment:
If I'm not mistaken, this function uses the cluster state that was generated at the beginning. If a user runs `kubeone apply -y` with changing the Kubernetes version, and with certificates that will expire soon, we can end up in the following situation:

1. Apply probes determine that the certificates will expire soon
2. kubeadm upgrades the cluster and renews all certificates
3. Since we're not refreshing the state after the upgrade, this predicate will be true as certificates were about to expire
4. KubeOne will show a message that the user should run `kubeone apply --force-upgrade`, however, certificates are already renewed by KubeOne when upgrading the cluster
    - This can be misleading to the user and the user will probably think that they need to run force upgrade, while that's not the case

This can also be a problem if the user runs `kubeone apply --force-upgrade` with changing the Kubernetes version. In that case, the certificates will be renewed two times.

Reply:
Running --force-upgrade with changed version is so pointless and marginal that I'd say that it doesn't matter that certs will be changed twice.